test/no-vars-test.js

```diff
@@ -0,0 +1,46 @@
+var module = require('module');
+
+for (var i = 0; i < 10; i++) {
+
+}
+
+for (i = 0; i < 10; i++) {
+}
+
+var shouldBeLet = 0;
+var shouldBeConst = 0;
+
+function mutate() {
+  shouldBeLet = 1;
+}
+
+var array = ['a', 'b', 'c', 'd'];
+
+for (var letter in array) {
+  console.log(letter);
+}
+
+var object = {
+  'a': 1,
+  'b': 2,
+  'c': 3,
+  'd': 4
+};
+
+for (var [key, value] of object.entries()) {
+  console.log(key, value);
+}
+
+var whileIterator = 10;
+while (whileIterator > 0) {
+  whileIterator--;
+}
+
+var z = 10;
+do {
+  var a = 0;
+  var b = 1;
+  a += b;
+  console.log(a, b);
+} while (z--);
+
```

Comment:
Yeah. I’m thinking about this. I added the below test for nested scopes and it appears the original comment wasn’t completely accurate. The below test works as expected, though shadowed variables currently don’t do the most optimal transform.

```
﻿
() => {
  var a = 1;

  return () => {
    return _ = _ => _ => _ => {
      var a = 3;
      return _ => _ => { a = 7; };
    }
  }()
}();
```

is being transformed to:

```
() => {
  let a = 1;

  return () => {
    return _ = _ => _ => _ => {
      let a = 3;
      return _ => _ => { a = 7; };
    }
  }()
}();
```